LibreNMS/OS/Lcos.php

```diff
@@ -0,0 +1,313 @@
+<?php
+/**
+ * Lcos.php
+ *
+ * Lancom LCOS
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2019 Vitali Kari
+ * @author     Vitali Kari <vitali.kari@gmail.com>
+ */
+
+namespace LibreNMS\OS;
+
+use LibreNMS\Device\WirelessSensor;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessFrequencyDiscovery;
+use LibreNMS\Interfaces\Polling\Sensors\WirelessFrequencyPolling;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessCapacityDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessNoiseFloorDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessPowerDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessCcqDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessRateDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessRssiDiscovery;
+use LibreNMS\OS;
+
+class Lcos extends OS implements
+    WirelessFrequencyDiscovery,
+    WirelessFrequencyPolling,
+    WirelessCapacityDiscovery,
+    WirelessNoiseFloorDiscovery,
+    WirelessPowerDiscovery,
+    WirelessCcqDiscovery,
+    WirelessRateDiscovery,
+    WirelessRssiDiscovery
+{
+    /**
+     * Discover wireless frequency.  This is in Hz. Type is frequency.
+     * Returns an array of LibreNMS\Device\Sensor objects that have been discovered
+     *
+     * @return array Sensors
+     */
+    public function discoverWirelessFrequency()
+    {
+        $data = snmpwalk_cache_oid($this->getDevice(), 'lcsStatusWlanRadiosEntryRadioChannel', [], 'LCOS-MIB');
+        $radios = $this->getCacheByIndex('lcsStatusWlanRadiosEntryIfc', 'LCOS-MIB');
+
+        $sensors = [];
+        foreach ($data as $index => $entry) {
+            $radio = $radios[$index];
+            if (isset($sensors[$radio])) {
+                continue;
+            }
+            for ($i = 0, $j = strlen($index); $i < $j; $i++) {
```

Comment:
Could you please put the conversion of the index to decimal into a function that would be called from each discovery function, instead of copy/pasting it ?

Thanx

Reply:
thank you for the hint, done.